study1/week3/climbing-stairs.java

```diff
@@ -0,0 +1,24 @@
+// https://leetcode.com/problems/climbing-stairs/
+// Time complexity: O(n), n: input value
+// Space complexity: O(1) because using only 3-size-array
+class Solution {
+    public int climbStairs(int n) {
+        if (n == 1) {   // edge cases
+            return 1;
+        } else if (n == 2) {
+            return 2;
+        }
+
+        int[] temp = new int[3];    // using dp
+        temp[0] = 1;
+        temp[1] = 2;
+
+        for (int i = 2; i < n; ++i) {
+            temp[2] = temp[0] + temp[1];
```

Comment:
A trick I learned from somewhere, of which you can use the remainder operator(`%`) to use to track the last element without having to copy back intermediate results. So you can do the following to make it simpler : 
```Java
class Solution {
    public int climbStairs(int n) {
        int[] temp = new int[]{1, 2};
        for (int i = 2; i < n; ++i) {
            temp[i % 2] = temp[0] + temp[1];
        }
        return temp[(n - 1) % 2];
    }
}
```